Xamarin.Forms.Core/ResourceDictionary.cs

```diff
@@ -256,5 +292,54 @@ void OnValuesChanged(params KeyValuePair<string, object>[] values)
 		}
 
 		event EventHandler<ResourcesChangedEventArgs> ValuesChanged;
+
+		[Xaml.ProvideCompiled("Xamarin.Forms.Core.XamlC.RDSourceTypeConverter")]
+		public class RDSourceTypeConverter : TypeConverter, IExtendedTypeConverter
+		{
+			object IExtendedTypeConverter.ConvertFromInvariantString(string value, IServiceProvider serviceProvider)
+			{
+				if (serviceProvider == null)
+					throw new ArgumentNullException(nameof(serviceProvider));
+
+				var targetRD = (serviceProvider.GetService(typeof(Xaml.IProvideValueTarget)) as Xaml.IProvideValueTarget)?.TargetObject as ResourceDictionary;
+				if (targetRD == null)
+					return null;
+
+				var rootObjectType = (serviceProvider.GetService(typeof(Xaml.IRootObjectProvider)) as Xaml.IRootObjectProvider)?.RootObject.GetType();
+				if (rootObjectType == null)
+					return null;
+
+				var lineInfo = (serviceProvider.GetService(typeof(Xaml.IXmlLineInfoProvider)) as Xaml.IXmlLineInfoProvider)?.XmlLineInfo;
+				var rootTargetPath = XamlResourceIdAttribute.GetPathForType(rootObjectType);
+				var uri = new Uri(value, UriKind.Relative); //we don't want file:// uris, aven if they start with '/'
```

Comment:
aven if => even if

Reply:
fixed. you should never read the comments. ever